components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.enrich.ui/src/main/resources/web/enrich-mediator/edit-mediator.jsp

```diff
@@ -210,41 +208,75 @@
         }
     }
 
-    function validateCombinations(currElem){
+    function validateCombinations(sourceId, targetId){
+        <fmt:message key="mediator.enrich.type.custom" var="custom"/>
+        <fmt:message key="mediator.enrich.type.body" var="body"/>
+        <fmt:message key="mediator.enrich.type.property" var="property"/>
+        <fmt:message key="mediator.enrich.type.envelope" var="envelope"/>
+
+        var common = ['${custom}', '${envelope}', '${body}', '${property}'];
+        var custom = ['${custom}', '${body}', '${property}'];
+        var envelope = ['${property}'];
+        var body = ['${custom}', '${property}'];
+
+        switch (sourceId.value) {
+            case 'property':
+            case 'inline':
+                targetId.options.length = 0;
```

Comment:
Is it possible to define this once at the top?

Reply:
Done